src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,39 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj == null) {
+            return false;
+        }
+
+        if (obj.getClass().equals(Rectangle.class)) {
+            Rectangle rect = (Rectangle) obj;
+
+            return (this.getWidth() == null && ((Rectangle) obj).getWidth() == null
+                    || width.equals(rect.getWidth()))
+                    && (this.getLength() == null && ((Rectangle) obj).getLength() == null
+                    || length.equals(rect.getLength()))
+                    && (this.getColor() == null && ((Rectangle) obj).getColor() == null
+                    || color.equals(rect.getColor()));
```

Comment:
1) Should we use getters inside class?
2) You already defined rect object, no need repeat this in every condition;
3) Try to simplify this condition